pkg/machines/hostvmslist.jsx

```diff
@@ -275,21 +259,28 @@ VmBootOrder.propTypes = {
     vm: PropTypes.object.isRequired
 };
 
-const VmOverviewTab = ({ vm, config }) => {
+const VmOverviewTab = function ({ vm, config, dispatch }) {
     let providerContent = null;
     if (config.provider.VmOverviewColumn) {
         const ProviderContent = config.provider.VmOverviewColumn;
         providerContent = (<ProviderContent vm={vm} providerState={config.providerState}/>);
     }
 
+    const handleOpenModal = function () {
+        VCPUModal({
+            vm,
+            dispatch,
+        });
+    };
+
     return (<div>
         <table className='machines-width-max'>
             <tr className='machines-listing-ct-body-detail'>
                 <td className='machines-listing-detail-top-column'>
                     <table className='form-table-ct'>
                         <VmOverviewTabRecord descr={_("Memory:")}
                                              value={cockpit.format_bytes((vm.currentMemory ? vm.currentMemory : 0) * 1024)}/>
-                        <VmOverviewTabRecord id={`${vmId(vm.name)}-vcpus`} descr={_("vCPUs:")} value={vm.vcpus}/>
+                        <VmOverviewTabRecord id={`${vmId(vm.name)}-vcpus`} descr={_("vCPUs:")} value={<a id={`${vmId(vm.name)}-vcpus-count`} data-toggle="modal" data-target={`${vmId(vm.name)}-vcpu-modal`} onClick={handleOpenModal}>{vm.vcpus.count}</a>}/>
```

Comment:
Please split

Reply:
For readability, I suggest to move the inner `<a>` element outside and reference it here via variable.